crates/indexer/src/db/market.rs

```diff
@@ -2246,6 +2292,162 @@ impl IndexerDb for MarketDb {
         Ok(request_digests)
     }
 
+    async fn get_cycle_counts_pending(&self, limit: u32) -> Result<HashSet<B256>, DbError> {
+        let query =
+            "SELECT request_digest FROM cycle_counts WHERE cycle_status = 'PENDING' ORDER BY updated_at DESC LIMIT $1";
+
+        let rows = sqlx::query(query).bind(limit as i64).fetch_all(self.pool()).await?;
+
+        let mut request_digests = HashSet::new();
+        for row in rows {
+            let digest_str: String = row.try_get("request_digest")?;
+            let digest = match B256::from_str(&digest_str) {
+                Ok(d) => d,
+                Err(e) => {
+                    tracing::warn!("Failed to parse request_digest '{}': {}", digest_str, e);
+                    continue;
+                }
+            };
+            request_digests.insert(digest);
+        }
+
+        Ok(request_digests)
+    }
+
+    async fn get_cycle_counts_executing(
+        &self,
+        limit: u32,
+    ) -> Result<HashSet<CycleCountExecution>, DbError> {
+        let query = "SELECT request_digest, session_uuid FROM cycle_counts WHERE cycle_status = 'EXECUTING' ORDER BY updated_at DESC LIMIT $1";
+
+        let rows = sqlx::query(query).bind(limit as i64).fetch_all(self.pool()).await?;
+
+        let mut execution_info = HashSet::new();
+        for row in rows {
+            let digest_str: String = row.try_get("request_digest")?;
+            let digest = match B256::from_str(&digest_str) {
+                Ok(d) => d,
+                Err(e) => {
+                    tracing::warn!("Failed to parse request_digest '{}': {}", digest_str, e);
+                    continue;
+                }
+            };
+            let session_uuid: String = row.try_get("session_uuid")?;
+            execution_info.insert(CycleCountExecution { request_digest: digest, session_uuid });
+        }
+
+        Ok(execution_info)
+    }
+
+    async fn set_cycle_counts_executing(
+        &self,
+        execution_info: &HashSet<CycleCountExecution>,
+    ) -> Result<(), DbError> {
+        let execution_vec: Vec<&CycleCountExecution> = execution_info.iter().collect();
+
+        let current_timestamp =
+            std::time::SystemTime::now().duration_since(std::time::UNIX_EPOCH).unwrap().as_secs();
```

Comment:
We should use CURRENT_TIMESTAMP here and elsewhere. e.g.: https://github.com/boundless-xyz/boundless/blob/willpote/cluster-prio/crates/indexer/src/db/market.rs#L3238

Reply:
Currently done this way due to BIGINT type but we can revisit